docs: add config of en version (#454)
umijs/dumi: docs/config/index.md

order: 1
toc: menu
nav:

>   title: Configuration items

Reviewer: ```suggestion
  title: Config
```

  order: 3
---

> # Config

Reviewer: ```suggestion
# Configuration
```

Create a `.umirc.ts` or `config/config.ts` file in the project root directory to configure dumi:

```ts

> // Configuration file

Reviewer: ```suggestion
// Configuration content
```

export default {
  // Configuration items
};
```

Currently dumi supports the following configuration items.

## algolia

- Type: `Object`

> - Defaults: `null`

Reviewer: ```suggestion
- Default: `null`
```

你本地统一改下

- Details:

> Configure Algolia's [DocSearch](https://docsearch.algolia.com/) service, usually you will need to enable the automatic generation of sitemap.xml in order to access Algolia smoothly, refer to [Configuration Item-sitemap](#sitemap).

Reviewer: ```suggestion
Configure Algolia's [DocSearch](https://docsearch.algolia.com/) service, usually you will need to enable the automatic generation of sitemap.xml in order to access Algolia smoothly, refer to [Configuration Item - sitemap](#sitemap).
```

Example：

```js
{
  algolia: {
    apiKey: 'Your api key',
    indexName: 'dumi',
  }
}
```

## description

- Type: `String`
- Defaults: `null`
- Details:

The introduction of the configuration document will be displayed under the sidebar menu title, only available in `doc` mode.

## logo

- Type: `String`

> - Defaults: Umi's LOGO

Reviewer: ```suggestion
- Defaults: The LOGO of Umi
```

- Details: 

> LOGO of configuration document.

Reviewer: ```suggestion
To set the LOGO of the document.
```

## locales

- Type: `Array<[String, String]>`
- Default: `[['en-US', 'English'], ['zh-CN', '中文']]`
- Details:

The configuration is a two-dimensional array, and the first configuration will be the default locale of the site.

Each configuration is an array of length 2. The first value of the array represents the name of the locale, which will be used for splicing routing prefixes and detecting which locale the file name belongs to. The second value represents the label of the locale, which will be used Display options when switching languages.

The file name suffix of the default locale is optional. For example, in the default configuration, `index.md` and `index.en-US.md` are equivalent.

## mode

- Type: `doc | site`
- Default: `doc`
- Details:

It is used to set the presentation mode of documents. The default is document mode (left menu + right content). When configured as `site`, it can be seamlessly switched to site mode (navigation header + left menu + right content). If you want to display the text and order of the navigation menu items, you can refer to the `nav` configuration item in the frontmatter configuration.

> The effects of the two modes can be seen in [Guide-Multiple Presentation Modes](/guide/mode).

Reviewer: ```suggestion
The effects of the two modes can be seen in [Guide - Modes](/guide/mode).
```

> The effects of the two modes can be seen in [Guide-Multiple Presentation Modes](/zh-CN/guide/mode).

Author: 这个链接的内容没找到诶🤔

Reviewer 2: 这是我的锅，直接拷贝 1.0 的文档了，这个 commit 补充了中文版的内容，这个 PR 也辛苦更正下：https://github.com/umijs/dumi/commit/170205e5476d43d25a4464e8b94b544b6474e66c

Author: 最下面有个 主题列表 的链接要不要也改一下

> The effects of the two modes can be seen in [Guide-Multiple Presentation Modes](/zh-CN/guide/mode).

Reviewer: ```suggestion
The effects of the two modes can be seen in [Guide-Multiple Presentation Modes](/guide/mode).
```

## menus

- Type: `Object`
- Defaults: `Auto-generated menu`
- Details:

> This configuration item is used to customize the display of the side menu. It is currently only available in the `site` mode, divided into multi-language mode and single language mode. Please refer to [Guide-Configurable Side Menu](/zh-CN/) guide/control-menu-generate#Configure side menu).

Reviewer: ```suggestion
This configuration item is used to customize the display of the side menu. It is currently only available in the `site` mode, divided into multi-language mode and single language mode. Please refer to [Guide - Configurable side menu](/guide/control-menu-generate#configurate-side-menu).
```

> This configuration item is used to customize the display of the side menu. It is currently only available in the `site` mode, divided into multi-language mode and single language mode. Please refer to [Guide-Configurable Side Menu](/zh-CN/) guide/control-menu-generate#Configure side menu).

Reviewer: ```suggestion
This configuration item is used to customize the display of the side menu. It is currently only available in the `site` mode, divided into multi-language mode and single language mode. Please refer to [Guide - Configurable side menu](/guide/control-menu-generate#configurate-side-menu).
```

## navs

- Type: `Object | Array`
- Defaults: `Auto-generated navigation`
- Details:

> This configuration item is used to customize the display of the navigation bar. It is only available in the `site` mode. It can be divided into multi-language mode and single-language mode. For details, please refer to [Guide-Configuration Navigation](/zh-CN/guide/control- nav-generate#Configuration navigation).

Reviewer: ```suggestion
This configuration item is used to customize the display of the navigation bar. It is only available in the `site` mode. It can be divided into multi-language mode and single-language mode. For details, please refer to [Guide - Configuration navigation](/guide/control-nav-generate#configurate-navigation).
```

The two-level navigation menu can be nested in the following forms, and currently does not support more levels of nesting:

```js
export default {
  navs: [
    {
      title: 'I have secondary navigation',
      path: 'Link is optional',
      children: [
        { title: 'First item', path: 'https://d.umijs.org' },
        { title: 'Second item', path: '/guide' },
      ],
    },
  ],
};
```

## resolve

`resolve` is an `Object` type, used to configure dumi's resolution behavior, including the following configuration:

### includes

- Type: `Array<String>`
- Default: `['docs', 'src']` or `['docs', 'packages/pkg/src']`
- Details:

> Configure the document directory for dumi sniffing. Dumi will try to recursively find markdown files in the configured directory. The default values are the `docs` directory and the `src` directory (common projects). If the environment is the lerna project, the `src` directory will change It is the `packages/pkg/src` directory, and usually does not need to be configured, unless the automatic sniffing appears 『injury』.

Reviewer: ```suggestion
Configure the document directory for dumi sniffing. Dumi will try to recursively find markdown files in the configured directory. The default values are the `docs` directory and the `src` directory (common projects). If the environment is the lerna project, the `src` directory will change It is the `packages/pkg/src` directory, and usually does not need to be configured, unless the automatic sniffing appears 『injuryed』.
```

### previewLangs

- Type: `Array<String>`
- Defaults: `['jsx', 'tsx']`
- Details:

The configuration dumi will be converted to the code block rendered by the ReactComponent component by default. If you don't want to do any conversion, such as a pure site like Umi's official website, then set this item to an empty array.

## sitemap

- Type: `{ hostname: string, excludes?: string[] }`

> - Default: `null`

Reviewer: 这里倒是想起来 用 Default 了

Enable the automatic generation feature of `sitemap.xml`. The `hostname` configuration item is used to specify the domain name prefix of the URL, and the `excludes` configuration item is used to ignore certain routes that do not need to be included in the sitemap.

## title

- Type: `String`
- Defaults: `{package.name}`
- Details:

Configure the name of the document on the navigation bar or sidebar.

## themeConfig

- Type: `Object`
- Defaults: `{}`
- Details:

> It's used to configure the theme package currently used. The specific configuration items depend on which configuration the theme package provides. You can visit [Theme List](/docs/theme/index.md) to view the currently available themes.

Reviewer: ```suggestion
It's used to configure the theme package currently used. The specific configuration items depend on which configuration the theme package provides. You can visit [Theme list](/theme) to view the currently available themes.
```

### previewLangs

- Type: `Array<String>`
- Default: `['jsx', 'tsx']`
- Details:

The configuration dumi will be converted to the code block rendered by the ReactComponent component by default. If you don't want to do any conversion, such as a pure site like Umi's official website, then set this item to an empty array.

## sitemap

- Type: `{ hostname: string, excludes?: string[] }`
- Default: `null`

Enable the automatic generation feature of `sitemap.xml`. The `hostname` configuration item is used to specify the domain name prefix of the URL, and the `excludes` configuration item is used to ignore certain routes that do not need to be included in the sitemap.

## title

- Type: `String`
- Default: `{package.name}`
- Details:

Configure the name of the document on the navigation bar or sidebar.

## themeConfig

- Type: `Object`
- Default: `{}`
- Details:

> It's used to configure the theme package currently used. The specific configuration items depend on which configuration the theme package provides. You can visit [Theme list](/theme) to view the currently available themes.

Reviewer 2: @ajuner 是指这里吗，这里是对的吧？

Author: 我以为要link到/doc/theme/index下来着

Reviewer: 😂😂  这是 dumi 的路由，在网页上点的

Author: 😂反应过来了
